opentelemetry-otlp/src/exporter/mod.rs

```diff
@@ -28,18 +28,32 @@ pub const OTEL_EXPORTER_OTLP_PROTOCOL: &str = "OTEL_EXPORTER_OTLP_PROTOCOL";
 /// Compression algorithm to use, defaults to none.
 pub const OTEL_EXPORTER_OTLP_COMPRESSION: &str = "OTEL_EXPORTER_OTLP_COMPRESSION";
 
-#[cfg(feature = "http-proto")]
+#[cfg(all(
+    feature = "http-json",
+    not(all(feature = "grpc-tonic", feature = "http-proto"))
+))]
+/// Default protocol, using http-json.
+pub const OTEL_EXPORTER_OTLP_PROTOCOL_DEFAULT: &str = OTEL_EXPORTER_OTLP_PROTOCOL_HTTP_JSON;
+#[cfg(all(
+    feature = "http-proto",
+    not(all(feature = "grpc-tonic", feature = "http-json"))
+))]
 /// Default protocol, using http-proto.
 pub const OTEL_EXPORTER_OTLP_PROTOCOL_DEFAULT: &str = OTEL_EXPORTER_OTLP_PROTOCOL_HTTP_PROTOBUF;
-#[cfg(all(feature = "grpc-tonic", not(feature = "http-proto")))]
-/// Default protocol, using grpc as http-proto feature is not enabled.
+#[cfg(all(
+    feature = "grpc-tonic",
+    not(all(feature = "http-proto", feature = "http-json"))
+))]
+/// Default protocol, using grpc
 pub const OTEL_EXPORTER_OTLP_PROTOCOL_DEFAULT: &str = OTEL_EXPORTER_OTLP_PROTOCOL_GRPC;
-#[cfg(not(any(any(feature = "grpc-tonic", feature = "http-proto"))))]
+
+#[cfg(not(any(feature = "grpc-tonic", feature = "http-proto", feature = "http-json")))]
 /// Default protocol if no features are enabled.
 pub const OTEL_EXPORTER_OTLP_PROTOCOL_DEFAULT: &str = "";
```

Comment:
@TommyCpp is this a valid default? Should this match the default feature enabled in Cargo.toml instead?

Reply:
No it's not. The `default` feature in Cargo.toml contains `grpc-tonic` so we shouldn't encounter unless users explicitly disable the default feature and didn't enable anything else.

Regardls